src/experimental/ProbabilisticGraphicalModels/conditioning.jl

```diff
@@ -29,29 +33,57 @@ function condition!(
         else
             bn.is_observed[id] = true
         end
-        bn.values[name] = value
+        new_evaluation_env = merge(new_evaluation_env, (name => value,))
     end
-    return bn
+    return BangBang.setproperties!!(bn; evaluation_env=new_evaluation_env)
 end
 
+"""
+    decondition(bn::BayesianNetwork{V}) where {V}
+
+Remove all conditioning from the Bayesian Network.
+"""
 function decondition(bn::BayesianNetwork{V}) where {V}
     conditioned_variables_ids = findall(bn.is_observed)
     return decondition(bn, bn.names[conditioned_variables_ids])
 end
 
+"""
+    decondition!(bn::BayesianNetwork{V}) where {V}
+
+Mutating version of [`decondition`](@ref).
+"""
 function decondition!(bn::BayesianNetwork{V}) where {V}
     conditioned_variables_ids = findall(bn.is_observed)
     return decondition!(bn, bn.names[conditioned_variables_ids])
 end
 
+"""
+    decondition(bn::BayesianNetwork{V}, variables::Vector{V}) where {V}
+
+Remove conditioning from a subset of variables in the Bayesian Network.
+"""
 function decondition(bn::BayesianNetwork{V}, variables::Vector{V}) where {V}
     is_observed = copy(bn.is_observed)
-    values = copy(bn.values)
-    bn_new = BangBang.setproperties!!(bn; is_observed=is_observed, values=values)
+    evaluation_env = NamedTuple(
```

Comment:
`evaluation_env` contains **all** the values always, no need to remove variables during unconditioning